pipelines/validate_run.py

```
@@ -74,13 +100,252 @@ def check_spike_order(events, where):
+            if not isinstance(val, str) or val not in ALLOWED_SIM_OR_REAL:
+                errs.append(
+                    f"{where}: state.sim_or_real must be one of {sorted(ALLOWED_SIM_OR_REAL)}"
+                )
+            # Explicit 'real' rejection (case-insensitive substring)
+            if isinstance(val, str) and "real" in val.lower():
+                errs.append(f"{where}: state.sim_or_real must not be 'real' (use 'designed')")
+        else:
+            # For v2 thalamic shapes, sim_or_real is mandatory; flag missing
+            # only when state looks like a thalamic state (has any expected key)
+            # to avoid false positives on unrelated objects.
+            if any(k in state for k in ("episode_id", "domain", "t0_us", "sim_or_real")):
```

Ruff (0.16.1)

[warning] Use `elif` instead of `else` then `if`, to reduce indentation

Convert to `elif`

(PLR5501)